src/aws_credentials_file.erl

```diff
@@ -26,19 +37,29 @@
         {error, any()} | {ok, aws_credentials:credentials(), 'infinity'}.
 fetch(Options) ->
     FilePath = get_file_path(Options),
-    ConfigPath = does_credentials_file_exist(FilePath, "config"),
-    case does_credentials_file_exist(FilePath, "credentials") of
+    ConfigPath = does_credentials_file_exist(FilePath, config),
+    case does_credentials_file_exist(FilePath, credentials) of
         {error, Error} ->
             {error, {credentials_file_does_not_exist, Error}};
         CredentialsPath ->
             CredFile = parse_credentials_file(CredentialsPath, Options),
             maybe_add_region(CredFile, ConfigPath, Options)
     end.
 
--spec does_credentials_file_exist(string(), string()) -> {error, any()} | string().
-does_credentials_file_exist({error, _} = Error, _FileName) -> Error;
-does_credentials_file_exist(Path, FileName) ->
-    check_path_exists(Path ++ FileName).
+-spec does_credentials_file_exist(string(), atom()) -> {error, any()} | string().
+does_credentials_file_exist({error, _} = Error, _File) -> Error;
+does_credentials_file_exist(Path, credentials) ->
+    maybe_path_from_env("AWS_SHARED_CREDENTIALS_FILE", check_path_exists(Path ++ "credentials"));
+does_credentials_file_exist(Path, config) ->
+    maybe_path_from_env("AWS_CONFIG_FILE", check_path_exists(Path ++ "config")).
+
+-spec maybe_path_from_env(string(), string()) -> {error, any()} | string().
+maybe_path_from_env(EnvVar, FilePath) ->
+    case {os:getenv(EnvVar), FilePath} of
+        {false, {error, _} = Error} -> Error;
+        {false, Path} -> Path;
+        {EnvPath, _} -> check_path_exists(EnvPath)
```

Comment:
Aren't you missing a case clause here where both the ENV and `FilePath` exists?

Reply:
Oh, it makes sense. I think it should be:

```erlang
    case {os:getenv(EnvVar), FilePath} of
        {false, {error, _} = Error} -> Error;
        {_, Path} -> Path;
        {EnvPath, _} -> check_path_exists(EnvPath)
    end.
```

WDYT?

Reply:
Sounds good! :+1:

Reply:
If I try to use the condition I suggested, we will never get from env. Or my test case is using a bad way to handle it or it's not working as I expected with:

```erlang
    case {os:getenv(EnvVar), FilePath} of
        {false, {error, _} = Error} -> Error;
        {false, Path} -> Path;
        {_, Path} when not is_tuple(Path) -> Path;
        {EnvPath, _} -> check_path_exists(EnvPath)
    end.
```